vminfo_parser/vminfo_parser.py

```diff
@@ -94,6 +94,54 @@ def add_extra_columns(self: t.Self) -> None:
         else:
             print("All columns already exist")
 
+    def add_site_specific_columns(self: t.Self) -> pd.DataFrame:
+        """
+        Adds site-specific columns to the DataFrame by aggregating resource usage metrics.
+        This function groups the data by site name and calculates the total memory, disk, and CPU usage for each site.
+
+        Args:
+            None: This method does not take any arguments.
+
+        Returns:
+            pd.DataFrame: A DataFrame containing the aggregated resource usage for each site, with renamed columns for clarity.
+
+        Examples:
+            site_usage_df = add_site_specific_columns()
+        """
+        site_columns = ["Site_RAM_Usage", "Site_Disk_Usage", "Site_CPU_Usage", "Site_VM_Count"]
+
+        # Check if all site-specific columns already exist
+        if all(col in self.df.columns for col in site_columns):
+            print("Site-specific columns already exist in the DataFrame.")
+            return
+
+        # Get the column names from the column_headers dictionary
+        memory_col = self.column_headers["vmMemory"]
+        disk_col = self.column_headers["vmDisk"]
+        cpu_col = self.column_headers["vCPU"]
+        unit_type = self.column_headers["unitType"]
+
+        if unit_type == "MB":
+            # If the disk and ram are in MB, convert the memory to GiB
+            # convert the disk to TiB
+            self.df[memory_col] = np.ceil(self.df[memory_col] / 1024).astype(int)
+            self.df[disk_col] = np.ceil(self.df[disk_col] / 1024 / 1024).astype(int)
+        elif unit_type == "GB":
+            # If the unit type is GB, we don't need to convert the ram
+            # convert the disk to TiB
+            self.df[disk_col] = np.ceil(self.df[disk_col] / 1024).astype(int)
+        elif unit_type != "GB":
+            raise ValueError(f"Unexpected unit type: {unit_type}")
```

Comment:
given that we are ultimately returning a new dataframe, do we want to be modifying the original dataframe as a side effect?  IF we do need to modify and store the change to the original dataframe,  that should probably be in a different function that is intended for that purpose.

Reply:
you are correct. I am now taking a copy of the original df